app/src/main/java/com/example/superimagesviewer/repository/MosaicRepository.kt

```diff
@@ -79,8 +110,9 @@ class MosaicRepository(application: Application) {
                     .build()
                 val drawable = imageLoader.execute(coilRequest).drawable
                 if (drawable != null) {
-                    drawables.add(drawable)
-                    mosaicsList.postValue(drawables)
+                    drawables.add(InstagramImage(drawable, imageUploadData))
+                    drawables.sortByDescending { it.uploadDataTime }
```

Comment:
This is kind of Business logic, it should be passed at ViewModel level

Reply:
done